### DISCUSSION TITLE
Interested in OGC Features API Support for GSoC 2022

### DISCUSSION THREAD

**Discussion by DummyPuppy** [koordinates/kart#608] [Google Soc / SoD]
Dear Kart Community,
Hi! This is Chloe Zeng, a master student from Carnegie Mellon University, and I am honored to join your team if I have the chance! I would like to work on the project "OGC Features API Support" for GSoC 2022, for which I believe I have the corresponding skill sets. I have had experiences in developing APIs in Java, handling HTTP requests in Python and supporting data processing. I am looking forward to integrating Kart into OAPIF to support more request needs from the websites. For now, I would like to know more about the project, and hopefully share some ideas on how to achieve certain deliverables!

Thank you all for your time!

**Comment by DummyPuppy** [comment]
And greeting to prospective mentors @rcoup  @craigds , I would like to discuss more about the project and what you expect of me!

**Comment by rcoup** [comment]
Hi Chloe, 

First step is to read through the materials from the ideas page and the associated links, then play with OGC API (there are several servers on the web — eg: [pygeoapi demos](https://demo.pygeoapi.io/)) and try and understand how the pieces fit together. Then your proposal needs to show how you could achieve the goal. For example, most of the pygeoapi backends are databases/equivalents, which is somewhat different to how Kart works.

The additional thing we'd like to see some thinking on are some concepts for how versioning can be incorporated into the APIs _in their existing form_. This could mean endpoints exposing specific commits (branches, tags); and maybe endpoints exposing the _version history_ as data.

**Reply by DummyPuppy** [reply]
Thank you, and I would look through them closer! Since the deadline is closing, I have to submit the proposal in time then, and I would catch up with all the detail here!
